spring-kafka/src/test/java/org/springframework/kafka/support/serializer/RetryingDeserializerTests.java

```diff
@@ -54,6 +55,39 @@ void testRetry() {
 		rdes.close();
 	}
 
+	@Test
+	void retryDeserializerWithRecoveryCallback() throws InterruptedException {
+		NonRecoverableDeser delegate = new NonRecoverableDeser();
+		RetryingDeserializer<String> rdes = new RetryingDeserializer<>(delegate, new RetryTemplate());
+		rdes.setRecoveryCallback(context -> {
+			delegate.latch.countDown();
```

Comment:
You can just sent here a result of `mock()` and then verify in the end that it was called.
There is no any race conditions: everything in the test is called sequentially.